internal/factory/container/device_linux.go

```diff
@@ -64,12 +75,13 @@ func (c *container) specAddHostDevicesIfPrivileged(privilegedWithoutHostDevices
 
 	for _, hostDevice := range hostDevices {
 		rd := rspec.LinuxDevice{
-			Path:  hostDevice.Path,
-			Type:  string(hostDevice.Type),
-			Major: hostDevice.Major,
-			Minor: hostDevice.Minor,
-			UID:   &hostDevice.Uid,
-			GID:   &hostDevice.Gid,
+			Path:     hostDevice.Path,
+			Type:     string(hostDevice.Type),
+			Major:    hostDevice.Major,
+			Minor:    hostDevice.Minor,
+			FileMode: getDeviceFileMode(hostDevice.Path),
```

Comment:
I think you can use [hostDevice.File](https://pkg.go.dev/github.com/opencontainers/cgroups/devices/config#Device) instead